vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java

```diff
@@ -147,6 +200,45 @@ public CreationReturnValue createImageBlock(String slideId, byte[] image, String
         return returnValue;
     }
 
+    /*
+     * (non-Javadoc)
+     * 
+     * @see edu.asu.diging.vspace.core.services.impl.IContentBlockManager#
+     * createVideoBlock(java.lang.String, java.util.Arrays, java.lang.String)
+     */
+    @Override
+    public CreationReturnValue createVideoBlock(String slideId, byte[] video, Long size, String fileName, String url, Integer contentOrder, String title)
+            throws VideoCouldNotBeStoredException {
+        ISlide slide = slideManager.getSlide(slideId);
+        CreationReturnValue returnValue = new CreationReturnValue();
+        returnValue.setErrorMsgs(new ArrayList<>());
+        IVideoBlock vidBlock = null;
+        IVSVideo slideContentVideo = storeVideo(video, size, fileName, url, contentOrder, title);
+        vidBlock = videoBlockFactory.createVideoBlock(slide, slideContentVideo);
+        vidBlock.setContentOrder(contentOrder);
+        VideoBlock videoBlock = videoBlockRepo.save((VideoBlock) vidBlock);
+        returnValue.setElement(videoBlock);
+        return returnValue;
+    }
+    
+    private IVSVideo storeVideo(byte[] video, Long size, String fileName, String url, Integer contentOrder, String title) 
+        throws VideoCouldNotBeStoredException {
+        IVSVideo slideContentVideo = null;
+        if( !((url == null || url.equals("")) && video == null))
+        {
+            if (url == null || url.equals("")) {
```

Comment:
that's an odd if condition, why not `if video != null`? More straight forward and easier to understand.